app/code/Magebit/Faq/Controller/Adminhtml/Question/Save.php

```diff
@@ -0,0 +1,133 @@
+            ?: \Magento\Framework\App\ObjectManager::getInstance()->get(QuestionFactory::class);
+        $this->questionRepository = $questionRepository
+            ?: \Magento\Framework\App\ObjectManager::getInstance()->get(QuestionRepositoryInterface::class);
+
+        parent::__construct($context);
+    }
+
+    /**
+     * Save action
+     *
+     * @SuppressWarnings(PHPMD.CyclomaticComplexity)
+     * @return \Magento\Framework\Controller\ResultInterface
+     */
+    public function execute()
+    {
+        $resultRedirect = $this->resultRedirectFactory->create();
+        $data = $this->getRequest()->getPostValue();
+        if ($data) {
+            if (isset($data['status']) && $data['status'] === 'true') {
+                $data['status'] = Question::STATUS_ENABLED;
+            }
+            if (empty($data['id'])) {
+                $data['id'] = null;
+            }
+            $model = $this->questionFactory->create();
+            $id = $this->getRequest()->getParam('id');
+            if ($id) {
+                try {
+                    $model = $this->questionRepository->getById($id);
+                } catch (LocalizedException $e) {
+                    $this->messageManager->addErrorMessage(__('This question no longer exists.'));
+                    return $resultRedirect->setPath('*/*/');
+                }
+            }
+
+            $model->setData($data);
+
+            try {
+                $this->questionRepository->save($model);
+                $this->messageManager->addSuccessMessage(__('You saved the question.'));
+                $this->dataPersistor->clear('question_block');
+                return $this->processBlockReturn($model, $data, $resultRedirect);
+            } catch (LocalizedException $e) {
+                $this->messageManager->addErrorMessage($e->getMessage());
+            } catch (\Exception $e) {
+                $this->messageManager->addExceptionMessage($e, __('Something went wrong while saving the question.'));
+            }
+
+            $this->dataPersistor->set('question_block', $data);
+            return $resultRedirect->setPath('*/*/edit', ['id' => $id]);
+        }
+        return $resultRedirect->setPath('*/*/');
+    }
+
+    /**
+     * Process and set the question return
+     *
+     * @param \Magebit\Faq\Model\Question $model
+     * @param array $data
+     * @param \Magento\Framework\Controller\ResultInterface $resultRedirect
+     * @return \Magento\Framework\Controller\ResultInterface
+     */
+    private function processBlockReturn($model, $data, $resultRedirect)
+    {
+        $redirect = $data['back'] ?? 'close';
+
+        if ($redirect === 'continue') {
+            $resultRedirect->setPath('*/*/edit', ['id' => $model->getId()]);
+        } elseif ($redirect === 'close') {
+            $resultRedirect->setPath('*/*/index');
+        } elseif ($redirect === 'duplicate') {
+            $duplicateModel = $this->questionFactory->create(['data' => $data]);
```

Comment:
- The newly created code follows the best practices of Magento development and doesn't contain any junk files or code.